src/components/forms/TextArea.tsx

```diff
@@ -0,0 +1,126 @@
+import styled from '@emotion/styled';
+import { CSSProperties, TextareaHTMLAttributes } from 'react';
+
+import { FullSpinner } from '../index';
+
+import { useFieldsContext } from './context/FieldsContext';
+import {
+  RenderAddon,
+  RootInput,
+  GroupStyled,
+  LabelStyled,
+  LeadingAddonStyled,
+  LeadingInlineAddonStyled,
+  TrailingInlineAddonStyled,
+  TrailingAddonStyled,
+} from './styles';
+
+const TextAreaStyled = styled.textarea`
+  padding: 0;
+  flex: 1 1 0%;
+  border: none;
+  position: relative;
+  outline: none;
+`;
+
+export interface TextAreaProps
+  extends TextareaHTMLAttributes<HTMLTextAreaElement> {
+  variant?: 'default' | 'small';
+
+  leadingAddon?: RenderAddon;
+  trailingAddon?: RenderAddon;
+
+  help?: string;
+  error?: string;
+  valid?: true | string;
+
+  loading?: boolean;
```

Comment:
Loading can also be removed. Could be potentially useful but would need to be designed differently and we don't have a use case for now.